### ISSUE TITLE
[Bug] math.format precision 0

### ISSUE THREAD

**Issue by dkenul** [josdejong/mathjs#1337]
Credit to @jmm for this find.

**Example:**
```js
math.format(16.01, {
  notation: 'fixed',
  precision: 0
})
// actual output: "16.01"
// expected output: "16"
```

[Live Example](https://runkit.com/585075bc6eea770014d02340/5bfec8c3b7acbb001253923a)

**Source:**
A falsey check here:
https://github.com/josdejong/mathjs/blob/c1cceec6ca37637147c6163ba057bdf8c86bae41/src/utils/number.js#L147

**Solution:**
`typeof precision === 'number'` as seen elsewhere in the file.

I can submit a pull request for this if verified as a bug.

**Comment by jmm** [comment]
I ran into the issue, but thanks to @dkenul for spotting the likely culprit!

**Comment by harrysarson** [comment]
Looks like a bug to me 👍 nice catch, thanks

**Comment by dkenul** [comment]
Thanks for the confirmation @harrysarson . Created a PR.

**Comment by josdejong** [comment]
This issue should be fixed now in `v5.3.0`. Thanks again @dkenul.
